datadotworld/client/api.py

```diff
@@ -650,13 +652,8 @@ def sparql(self, dataset_key, query, desired_mimetype='application/json',
         >>> api_client.sparql_post('username/test-dataset',\
         >>> query) # doctest: +SKIP
         """
-        sparql_api_client = content_negotiating_api_client. \
-            ContentNegotiatingApiClient(
-                host=self._host,
-                header_name='Authorization',
-                header_value='Bearer {}'.format(self._config.auth_token),
-                default_mimetype=desired_mimetype)
-        sparql_api_client.user_agent = _user_agent()
+        sparql_api_client = self._content_negotiating_client
+        sparql_api_client.default_mimetype = desired_mimetype
```

Comment:
This is changing a shared instance. We don't want that.
If you want to share logic related to the instantiation of `ContentNegotiatingApiClient`, I suggest you consider a different approach using `functools.partial`.

Reply:
I've leveraged the use of `copy.copy` to create a copy of the instantiation.